scripts/prepare_network.py

```diff
@@ -237,16 +225,17 @@ def set_line_nom_max(n):
     for o in opts:
         m = re.match(r'^\d+seg$', o, re.IGNORECASE)
         if m is not None:
-            n = apply_time_segmentation(n, m.group(0)[:-3])
+            n = apply_time_segmentation(n, m.group(0)[:-3], solver_name=snakemake.config["solving"]["solver"]["name"])
             break
 
     for o in opts:
         if "Co2L" in o:
             m = re.findall("[0-9]*\.?[0-9]+$", o)
             if len(m) > 0:
-                add_co2limit(n, Nyears, float(m[0]))
+                co2limit=float(m[0])*snakemake.config['electricity']['co2base']
+                add_co2limit(n, Nyears, co2limit)
```

Comment:
The order of arguments is incorrect (due to changes to the add_co2limit function).